tests/examples/MQ/pixelDetector/test_PixelGeo.cxx

```diff
@@ -0,0 +1,60 @@
+/********************************************************************************
+ *   Copyright (C) 2024 GSI Helmholtzzentrum fuer Schwerionenforschung GmbH     *
+ *                                                                              *
+ *              This software is distributed under the terms of the             *
+ *              GNU Lesser General Public Licence (LGPL) version 3,             *
+ *                  copied verbatim in the file "LICENSE"                       *
+ ********************************************************************************/
+
+#include <PixelGeo.h>
+#include <catch2/catch_test_macros.hpp>
+#include <string>
+
+TEST_CASE("PixelGeo")
+{
+    PixelGeo geo;
+
+    SECTION("Constructor")
+    {
+        // Check that members are correctly set by constructor
+        std::string retVal = geo.GetName();
+        REQUIRE(retVal == "Pixel");
+        REQUIRE(geo.getMaxModules() == 10);
+    }
+
+    SECTION("getModuleName")
+    {
+        std::string retVal = geo.getModuleName(0);
+        REQUIRE(retVal == "Pixel1");
+
+        retVal = geo.getModuleName(1);
+        REQUIRE(retVal == "Pixel2");
+
+        retVal = geo.getModuleName(99);
+        REQUIRE(retVal == "Pixel100");
+    }
```

Comment:
This should use a [Catch2 generator](https://github.com/catchorg/Catch2/blob/devel/docs/generators.md), see my deduplication patch for an example.

Reply:
A simpler version of my example in the deduplication patch could be e.g. just:

```cpp
    SECTION("name generation")
    {
        const Int_t index = GENERATE(0, 1, 99);

        REQUIRE(geo.getModuleName(index) == fmt::format("{}{}", prefix, index));
        REQUIRE(geo.getEleName(index) == fmt::format("{}{}", prefix, index));
    }
```